module/zfs/arc.c

```diff
@@ -4242,6 +4244,7 @@ arc_evict_adj(uint64_t frac, uint64_t total, uint64_t up, uint64_t down,
 	int s = highbit64(total);
 	s = MIN(64 - s, 32);
 
+	ASSERT3U(frac, <=, 1ULL << 32);
```

Comment:
is the value 1ULL << 32 (needing 33 bits) really allowed?

Reply:
ok thanks, and sorry for the noise. I was wondering because #17210 mentioned "capped at 32 bit".